### PR TITLE
das: 0.3.8 -> 1.0.3

### PR DESCRIPTION
**Pull request by ocfox**
Diff: https://github.com/snovvcrash/DivideAndScan/compare/v0.3.8...v1.0.3

Fix https://hydra.nixos.org/build/249726566
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/da/das/package.nix [hunk 1]
```diff
@@ -5,26 +5,28 @@
...
+    hash = "sha256-3d2sYxht4AMxz9K7Hm9VYWJ0bTfVFeGZFC+wGwteXCQ=";
+    fetchSubmodules = true;
   };
 
+  nativeBuildInputs = [
+    python3.pkgs.poetry-core
+    python3.pkgs.pythonRelaxDepsHook
+  ];
+
   postPatch = ''
```
Comment (ocfox) [inline pkgs/by-name/da/das/package.nix:24]: pythonRelaxDepsHook not work.

File: pkgs/by-name/da/das/package.nix [hunk 2]
```diff
@@ -5,26 +5,28 @@
...
-  format = "pyproject";
+  version = "1.0.3";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "snovvcrash";
     repo = "DivideAndScan";
-    rev = "refs/tags/v${version}";
-    hash = "sha256-a9gnEBTvZshw42M/GrpCgjZh6FOzL45aZqGRyeHO0ec=";
+    rev = "v${version}";
```
Comment (fabaff) [inline pkgs/by-name/da/das/package.nix:14]: Revert this. Branch/tags confusion is an issue for the updates made by the bot.

```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/by-name/da/das/package.nix [hunk 3]
```diff
@@ -5,26 +5,28 @@
...
+    hash = "sha256-3d2sYxht4AMxz9K7Hm9VYWJ0bTfVFeGZFC+wGwteXCQ=";
+    fetchSubmodules = true;
   };
 
+  nativeBuildInputs = [
+    python3.pkgs.poetry-core
+    python3.pkgs.pythonRelaxDepsHook
+  ];
```
Comment (fabaff) [inline pkgs/by-name/da/das/package.nix:22]:
```suggestion
```

Can be removed, same content will be identical line 24-27 as `pythonRelaxDepsHook` is not needed.

File: pkgs/by-name/da/das/package.nix [hunk 4]
```diff
@@ -5,26 +5,28 @@
...
     substituteInPlace pyproject.toml \
       --replace 'networkx = "^2.8.4"' 'networkx = "*"' \
-      --replace 'pandas = "^1.4.2"' 'pandas = "*"'
+      --replace 'netaddr = "^0.8.0"' 'netaddr = "*"'
   '';
 
-  nativeBuildInputs = with python3.pkgs; [
-    poetry-core
-  ];
-
```
Comment (fabaff) [inline pkgs/by-name/da/das/package.nix:27]: Leave this at this position. It helps if the building blocks are more or less in order of the execution.

File: pkgs/by-name/da/das/package.nix [hunk 5]
```diff
@@ -38,14 +40,13 @@ python3.pkgs.buildPythonApplication rec {
     tinydb
   ];
 
-  pythonImportsCheck = [
-    "das"
-  ];
+  pythonImportsCheck = [ "das" ];
```
Comment (fabaff) [inline pkgs/by-name/da/das/package.nix:43]: This change is not needed.

File: pkgs/by-name/da/das/package.nix [hunk 6]
```diff
@@ -5,24 +5,24 @@
...
-  format = "pyproject";
+  version = "1.0.3";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "snovvcrash";
     repo = "DivideAndScan";
     rev = "refs/tags/v${version}";
-    hash = "sha256-a9gnEBTvZshw42M/GrpCgjZh6FOzL45aZqGRyeHO0ec=";
+    hash = "sha256-3d2sYxht4AMxz9K7Hm9VYWJ0bTfVFeGZFC+wGwteXCQ=";
```
Comment (fabaff) [inline pkgs/by-name/da/das/package.nix:15]: I'm getting a mismatch:

``bash
error: hash mismatch in fixed-output derivation '/nix/store/0w7lld3vrla8l63jkgaqjlgajpmah37r-source.drv':
         specified: sha256-3d2sYxht4AMxz9K7Hm9VYWJ0bTfVFeGZFC+wGwteXCQ=
            got:    sha256-WZmWpcBqxsNH96nVWwoepFhsvdxZpYKmAjNd7ghIJMA=
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 289654` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>das</li>
    <li>das.dist</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Change looks good to me.

**Pull request merged**